### PR TITLE
chore: reduce continuous integration time

### PR DESCRIPTION
**Pull request by tartavull**
We make a list of all the files that has being changed
And we only test for the current changes.

The idea is that when we modify the frontend, that cannot
break anything in the backend.
So we only run frontend tests and deployments, which saves
time.

The drawback is that if something is brokend in the backend,
and we only modified the frontend. Travis will succed but
there will be still problems there. So we need to be extra
carefull that the base is correct.
To aliviate this problem we set up a daily cron job that
will build and deploy everyting.

### CODE REVIEW

**Comment by tartavull** [comment]
What do you guys think?

**Comment by tartavull** [comment]
Current problem is that I'm comparing HEAD to HEAD^1.
But the you could push more than one commit at the time.

File: was_modified.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from __future__ import print_function
+import subprocess
+import sys
+import os.path
+
+# filepath starts from the root directory of the git project
+# with not trailing slash at the begining
+# the last -- it's no separte revisions from files
+files = subprocess.check_output(
+    ["git", "diff", "--name-only", "HEAD", "HEAD^1" , '--']).split()
```
Comment (nkemnitz) [inline was_modified.py]: > Current problem is that I'm comparing HEAD to HEAD^1.
> But the you could push more than one commit at the time.

That's too dangerous, in a few months nobody will remember this pitfall. 

You could try `$TRAVIS_COMMIT_RANGE`, but that still has issues with history rewrites: https://github.com/travis-ci/travis-ci/issues/2668
One part of that problem might be solved using the Github API, see https://github.com/nest/nest-simulator/pull/75

**Comment by jingpengw** [comment]
@Ignacio What is the problem of testing all of them?

**Comment by tartavull** [comment]
@jingpengw life is to short to wait for travis to finish testing

File: was_modified.py (hunk 1 above)
Comment (tartavull) [reply was_modified.py]: This is great! Thank you Nico! I did tried $TRAVIS_COMMIT_RANGE commit range and found the history rewrites problem the hard way.

Comment (tartavull) [reply was_modified.py]: This seems to be another reasonable approach
https://github.com/reactor/reactor-core/pull/655/files

**Comment by tartavull** [comment]
Some other interesting approach
https://github.com/korfuri/bazel-travis

**Comment by nkemnitz** [comment]
lgtm

**Pull request merged**